client/src/Main.elm

```diff
@@ -1277,6 +1365,63 @@ getAuthentication auth =
         }
 
 
+getVectors : Auth -> Cmd Msg
+getVectors auth =
+    case auth of
+        Authenticated usr ->
+            authenticatedGet usr.token "http://localhost:8000/vectors/" VectorsReceived vectorDecoder
+
+        _ ->
+            Cmd.none
+
+
+createVector : String -> Vector -> Cmd Msg
+createVector token vector =
+    case vector of
+        Level0Vec vec ->
+            Http.request
+                { method = "POST"
+                , headers = [ Http.header "Authorization" ("Bearer " ++ token) ]
+                , url = "http://localhost:8000/vectors/"
+                , body = Http.jsonBody (level0Encoder vec)
+                , expect = Http.expectJson Level0Created level0Decoder
+                , timeout = Nothing
+                , tracker = Nothing
+                }
```

Comment:
You can eliminate some of this unnecessary duplication by writing an `authenticatedPost` function.

Then, with a `vectorEncoder` function from the `Molecules` module the entire body of this function becomes:
```elm
createVector : Auth -> Cmd Msg
createVector auth vector =
    case auth of
        Authenticated user ->
            authenticatedPost user.token "http://localhost:8000/vectors/" VectorCreated vectorEncoder
```
or something like that

Reply:
Taken care of in the latest commit.